src/Caching/Impl/ModuleUniqueId.cs

```diff
@@ -57,75 +50,12 @@ internal static class ModuleUniqueId {
-                    // Example: 'requests' and 'requests-2.21.0.dist-info'.
-                    // TODO: for egg (https://github.com/microsoft/python-language-server/issues/196), consider *.egg-info
-                    var folders = fs.GetFileSystemEntries(versionFolder, "*-*.dist-info", SearchOption.TopDirectoryOnly)
-                        .Select(Path.GetFileName)
-                        .Where(n => n.StartsWith(moduleRootName, StringComparison.OrdinalIgnoreCase)) // Module name can be capitalized differently.
-                        .ToArray();
-
-                    if (folders.Length == 1) {
-                        var fileName = Path.GetFileNameWithoutExtension(folders[0]);
-                        var dash = fileName.IndexOf('-');
-                        var uniqueId = $"{moduleName}({fileName.Substring(dash + 1)})";
-                        _idCache[key] = uniqueId;
-                        return uniqueId;
-                    }
-                    // Move up if nothing is found.
-                    versionFolder = Path.GetDirectoryName(versionFolder);
-                }
-            }
-
-            var config = interpreter.Configuration;
-            if (moduleType == ModuleType.CompiledBuiltin || string.IsNullOrEmpty(filePath) || modulePathType == PythonLibraryPathType.StdLib) {
-                // If module is a standard library, unique id is its name + interpreter version.
-                return $"{moduleName}({config.Version.Major}.{config.Version.Minor})";
-            }
-
             var parent = moduleResolution.CurrentPathResolver.GetModuleParentFromModuleName(moduleName);
             if (parent == null) {
                 return moduleName;
             }
-
-            var hash = HashModuleFileSizes(parent);
-            // If all else fails, hash modules file sizes.
-            return $"{moduleName}.{(ulong)hash}";
-        }
-
-        private static long HashModuleFileSizes(IImportChildrenSource source) {
-            var hash = 0L;
-            var names = source.GetChildrenNames();
-            foreach (var name in names) {
-                if (source.TryGetChildImport(name, out var child)) {
-                    if (child is ModuleImport moduleImport) {
-                        if (moduleImport.ModuleFileSize == 0) {
-                            continue; // Typically test case, memory-only module.
-                        }
-                        hash = unchecked(hash * 31 ^ moduleImport.ModuleFileSize);
-                    }
-
-                    if (child is IImportChildrenSource childSource) {
-                        hash = unchecked(hash * 31 ^ HashModuleFileSizes(childSource));
-                    }
-                }
-            }
-
-            return hash;
+            var config = interpreter.Configuration;
+            return $"{moduleName}({config.Version.Major}.{config.Version.Minor})";
```

Comment:
I'm not really sure if going all the way to python-versions is a good idea either...

Reply:
Yes, we need to tell if module was updated. Now, version if typically reflected in the path like `dist.info` so adding hash of the path is cheap way. 

However, this does not cover case when module path does not change (such as when it does not have version), but content does.
Hashing module sizes of all the child modules is expensive, so here I am not yet sure what to do.

Reply:
Restored most of the original, instead of file system walk code look into path directly. Also caches unique id per triplet in order to avoid multiple calculations for separate calls. Better, but still not quite. Needs some more profiling. Some time is taken by cache writing, of course.
```
Analysis complete: 2987 modules in 73918.4 ms.
```
Uncached
```
Analysis complete: 2987 modules in 60060.78 ms.
```

Reply:
Delayed cache writing to after analysis, look OK.